expertise/service/utils.py

```diff
@@ -158,3 +159,234 @@ def get_user_id(openreview_client):
+            camel_str = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', camel_str)
+            return re.sub('([a-z0-9])([A-Z])', r'\1_\2', camel_str).lower()
+        
+        # Get the name of the job
+        self.name = _get_field_from_request('name')
+
+        # These are automatically provided by the server
+        self.user_id = _get_field_from_request('user_id')
+        self.job_id = _get_field_from_request('job_id')
+        self.token = _get_field_from_request('token')
+        self.baseurl = _get_field_from_request('baseurl')
+        self.baseurl_v2 = _get_field_from_request('baseurl_v2')
+
+        # Validate entityA and entityB
+        entity_a = _get_field_from_request('entityA')
+        entity_b = _get_field_from_request('entityB')
+
+        _load_entity_a(entity_a)
+        _load_entity_b(entity_b)
+
+        # Optionally check for model object
+        self.model = request.pop('model', {})
+
+        # Check for empty request
+        if len(request.keys()) > 0:
+            raise openreview.OpenReviewException(f"Bad request: unexpected fields in {root_key}: {list(request.keys())}")
+    
+    def _load_entity(self, entity_id, source_entity, target_entity):
+        '''Load information from an entity into the config'''
+        def _get_from_entity(key):
+            return _get_required_field(source_entity, entity_id, key)
+
+        type = _get_from_entity('type')
+        target_entity['type'] = type
+        # Handle type group
+        if type == 'Group':
+            if 'memberOf' in source_entity.keys():
+                target_entity['memberOf'] = _get_from_entity('memberOf')
+                # Check for optional expertise field
+                if 'expertise' in source_entity.keys():
+                    target_entity['expertise'] = source_entity.pop('expertise')
+            else:
+                raise openreview.OpenReviewException(f"Bad request: no valid {type} properties in {entity_id}")
+        # Handle type note
+        elif type == 'Note':
+            if 'invitation' in source_entity.keys() and 'id' in source_entity.keys():
+                raise openreview.OpenReviewException(f"Bad request: only provide a single id or single invitation in {entity_id}")
+
+            if 'invitation' in source_entity.keys():
+                target_entity['invitation'] = _get_from_entity('invitation')
+            elif 'id' in source_entity.keys():
+                target_entity['id'] = _get_from_entity('id')
+            else:
+                raise openreview.OpenReviewException(f"Bad request: no valid {type} properties in {entity_id}")
+        else:
+            raise openreview.OpenReviewException(f"Bad request: invalid type in {entity_id}")
+
+        # Check for extra entity fields
+        if len(source_entity.keys()) > 0:
+            raise openreview.OpenReviewException(f"Bad request: unexpected fields in {entity_id}: {list(source_entity.keys())}")
+
+
+class ServerConfig(object):
```

Comment:
I'm realizing that this configuration is attached to the job that is running, why don't we call it `JobConfig` instead?

Reply:
I agree that would be a more fitting name for this class